### ISSUE TITLE
Rendering not working (Failed to Render: ERROR 1 Syntax Error? "Some diagram description contains errors")

### ISSUE THREAD

**Issue by RobiIhs** [Fruchtzwerg94/PlantUmlViewer#4]
Hello,
first of all, thank you for your effort.
I tried the plugin, downloading PlantUmlViewer_v1.1.1.5_x64.zip and placing it in the plugin folder (the plugin manager doesn't show the plugin in the list).

Whatever the code may be I always receive the same error:

> Failed to Render
> ERROR 
> 1 
> Syntax Error? 
> Some diagram description contains errors"

Only once it succeeded and I saw the image in the Plant UML panel.

Using the .jar from _cmd_ the .png is produced correctly:

> C:\Program Files\Notepad++\plugins\PlantUMLViewer>java -jar plantuml-1.2022.5.jar "C:\Users\..\Documents\UML\npp_plantum_example.pu"

This is the code I used to produce the .png with previous command:

```
@startuml

title Relationships - Class Diagram


class Dwelling {
  +Int Windows
  +void LockTheDoor()
}

class Apartment
class House
class Commune
class Window
class Door

Dwelling <|-down- Apartment: Inheritance
Dwelling <|-down- Commune: Inheritance
Dwelling <|-down- House: Inheritance
Dwelling "1" *-up- "many" Window: Composition
Dwelling "1" *-up- "many" Door: Composition

@enduml
```

I updated Notepad++, tried previous version of the plugin, ...

Thanks,
Roberto

**Comment by RobiIhs** [comment]
Ok, thanks for your quick answer.
I thought I had the last version of NPP but the updater stopped to 8.3.3.
However also 8.4.2, with other two plugins was raising the same error.

So ...

I started with a plain fresh npp.8.4.2.portable.x64.zip.
I installed "PlantUML Viewer" plugin with plugin manager: the rendering worked.
I added "ComparePlugin" plugin with plugin manager: the rendering worked.
I added "HexEditor" plugin with plugin manager: the rendering raised the error.

```
Failed to Render
ERROR
1
Syntax Error?
```

Let me tell you that this is not the 1st time that "HexEditor" causes me pains.
It already happened in the past but that plugin is really helpful and from time to time I need it.
For the moment, I'm going to remove it in order to go on playing with "PlantUML Viewer".

To make "PlantUML Viewer" work again I tried to remove "HexEditor"
- plugin manager wasn't able to remove it
- removing "HexEditor" from the plugin folder manually didn't solve too, the error was still appearing

Finally, in a new installation, I selected all 3 plugins _together_ in the plugin manager, installing them _together_: _all of them_ are working properly. I don't know if it's luck.

Thanks,
Roberto

**Comment by RobiIhs** [comment]
Hi Philipp,
I'm sorry, I disappeared for a while.
I think today I discovered the problem.
I had the same error on a portable installation I already tested before.
I was wandering why it wasn't working anymore of course.

The fact is that I use very frequently the double "View" (left-right). Normally I keep the "secondary" on the right.
Well I go to the point: even if the cursor is always in the plant-uml source tab, the rendering works only when it's in the left-main "View".
So, I suspected that when the plant-uml source tab is on the "other" right View, the plugin is reading from the last current file of the main left View and since that file is general NOT plant-uml file the syntax wasn't respected, obviously.
I confirmed this using two plant-uml files, one on the left and one on the right. The rendering always uses the one on the left, ignoring the position of the cursor.

I'm sorry for your wasted time.
Cheers,
Roberto
